tests/providers/class-two-factor-provider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Test Two Factor Provider
+ *
+ * @package Two_Factor
+ */
+
+/**
+ * Class Tests_Two_Factor_Provider
+ *
+ * @package Two_Factor
+ * @group providers
+ */
+class Tests_Two_Factor_Provider extends WP_UnitTestCase {
+	/**
+	 * @covers Two_Factor_Provider::get_code
+	 */
+	function test_get_code() {
+		$code = Two_Factor_Provider::get_code( 3, '1' );
+		$this->assertEquals( '111', $code );
+
+		$code = Two_Factor_Provider::get_code( 8, '1' );
+		$this->assertEquals( '11111111', $code );
+
+		$code = Two_Factor_Provider::get_code( 8, 'A' );
+		$this->assertEquals( 'AAAAAAAA', $code );
```

Comment:
```suggestion
		$this->assertEquals( 'AAAAAAAA', $code );
		
		$code = Two_Factor_Provider::get_code( 30, array( 'A', 'B', 'C' ) );
		$this->assertSame( 1, preg_match( '/^[ABC]*$/', $code ) );
```

This tests the `$chars = implode( '', $chars );` statement.

Reply:
Good call, I extended that to also validate that the generated code is 30char as requested, and an additional test that validates non-1-char-length strings.

Reply:
👍🏻